foundation/device/device-id/src/main/kotlin/com/pingidentity/device/id/KeyManager.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.device.id
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import java.math.BigInteger
+import java.security.Key
+import java.security.KeyPairGenerator
+import java.security.KeyStore
+import java.util.Random
+import javax.security.auth.x500.X500Principal
+
+/**
+ * Utility class for managing cryptographic keys in the Android KeyStore.
+ *
+ * This class provides functionality to:
+ * - Generate and retrieve RSA key pairs
+ * - Check for the existence of keys
+ * - Remove keys from the KeyStore
+ *
+ * The keys generated by this manager:
+ * - Use 2048-bit RSA key pairs
+ * - Are stored in the system's hardware-backed KeyStore when available
+ * - Persist across app restarts
+ */
+object KeyManager {
+
+    /** The name of the Android KeyStore provider */
+    private const val ANDROID_KEYSTORE: String = "AndroidKeyStore"
+
+    /** The Common Name used in the certificate's Distinguished Name */
+    private const val CN_PING_IDENTITY = "CN=PINGIDENTITY"
+
+    /** The size in bits of the RSA key to generate */
+    private const val KEY_SIZE = 2048
+
+    /**
+     * Returns a public key for the specified alias.
+     * If a key with the given alias doesn't exist, a new key pair is generated.
+     *
+     * @param alias The alias under which the key is stored in the KeyStore
+     * @return The public key corresponding to the alias
+     */
+    internal fun identifierKey(alias: String): Key {
+        val keyStore = this.keyStore
+        return if (!containsKey(alias)) {
+            KeyPairGenerator.getInstance(KeyProperties.KEY_ALGORITHM_RSA, ANDROID_KEYSTORE).apply {
+                initialize(getSpec(alias))
+            }.generateKeyPair().public
+        } else {
+            keyStore.getCertificate(alias).publicKey
+        }
+    }
+
+    /**
+     * Checks if a key with the specified alias exists in the KeyStore.
+     *
+     * @param alias The alias to check
+     * @return true if the key exists, false otherwise
+     */
+    internal fun containsKey(alias: String): Boolean {
+        return keyStore.containsAlias(alias)
+    }
+
+    /**
+     * Removes a key with the specified alias from the KeyStore.
+     *
+     * @param alias The alias of the key to remove
+     */
+    internal fun removeKey(alias: String) {
+        keyStore.deleteEntry(alias)
```

Comment:
Does keystore throw? I suggest using a `try/catch` on those and handle any exceptions if possible to avoid random Android devices crashes on these

Reply:
Same for `identifierKey` and `containsKey`

Reply:
The runtime exception will propagate to the caller and eventually be wrapped in a DeviceIdentifierException. However, developers still need to handle the exception to prevent crashes, unless a fallback or default value is provided.